src/lib/widget/_utilities/api.utilities.ts

```diff
@@ -0,0 +1,232 @@
+import type { z } from 'zod';
+
+// Import the stores for initialization
+import configure from '$lib/sdk.config';
+
+// Import store types
+import type { JourneyOptions, Modal, Response } from '../interfaces';
+import type { JourneyStore, JourneyStoreValue } from '$journey/journey.interfaces';
+import type { OAuthStore, OAuthTokenStoreValue } from '$lib/oauth/oauth.store';
+import type { partialConfigSchema } from '$lib/sdk.config';
+import type { UserStore } from '$lib/user/user.store';
+import type { HttpClientRequestOptions } from '@forgerock/javascript-sdk/lib/http-client';
+import HttpClient from '@forgerock/javascript-sdk/lib/http-client';
+import {
+  Config,
+  FRUser,
+  SessionManager,
+  TokenManager,
+  UserManager,
+  type GetTokensOptions,
+} from '@forgerock/javascript-sdk';
+import { get } from 'svelte/store';
+
+export function widgetApiFactory(modal: Modal) {
```

Comment:
I'm waffling on how i feel about this. I kind of think this should be a `class`. However if we want to stay "functional" we can probably use some other tools from that realm. 

I am hesitant to have a stateful factory, it just feels odd to me. This seems like a good use of a state machine though. Have we considered this as an option? 

Here are ways I see us building this without what I feel is a "stateful" factory since we have to keep the response value in the factory function outside of the configuration object.

1. a class, as this is more or less what we've built here.
2. we can use something _like_ a state monad (sorry for monad word).
3. a state machine (like xstate type implementation).
4. We can keep this implementation.

this implementation feels just an awful lot like a class, so i'm questioning why even take this approach versus a class?

To be clear, there isn't anything inherently wrong with the code. However, if the intent of the code is "to be functional" I think we are looking at what is equivalent to a `class` without the `new` keyword.

Reply:
I had the same thoughts as well. I actually started writing a class implementation of this. I had a few issues with it after some experimentation. Here they are in bullet form, but I can go into more details, if desired:

What I wanted to avoid ...

1. prototype pollution or misc. state hanging into the exported API
2. worrying about the `this` reference
3. worrying about method versus function execution context
4. leaking state information
5. cumbersome object shape influenced by the class pattern

What I did want ...

1. be able to export the created object as-is for a clean API
2. object methods that have a persistent, explicit state reference, rather than an implicit one
3. no-nonsense, hierarchical object creation free from any restriction or hacking

It's worth noting that the block of code within the `<script type="module">` annotation is only run once when the app bootstraps itself. It is not rerun unless the browser is reloaded. Reference: https://svelte.dev/docs#component-format-script-context-module.

Now, with that said, I'm completely open to the use of classes if you feel it's still the best way forward, but I would like to ask for your help. I am terrible at writing classes, and the class feature itself may have improved since I last used them, so your help in refactoring this to use them would be awesome.